extension/js/content_scripts/webmail/setup-webmail-content-script.ts

```diff
@@ -286,16 +286,26 @@ export const contentScriptSetupIfVacant = async (webmailSpecific: WebmailSpecifi
     }
   };
 
-  const startPullingKeysFromEkm = async (acctEmail: string, clientConfiguration: ClientConfiguration, factory: XssSafeFactory, ppEvent: { entered?: boolean }) => {
+  const startPullingKeysFromEkm = async (
+    acctEmail: string,
+    clientConfiguration: ClientConfiguration,
+    factory: XssSafeFactory,
+    ppEvent: { entered?: boolean },
+    completion: () => void | Promise<void>
+  ) => {
     if (clientConfiguration.usesKeyManager()) {
       const idToken = await InMemoryStore.get(acctEmail, InMemoryStoreKeys.ID_TOKEN);
       if (idToken) {
         const keyManager = new KeyManager(clientConfiguration.getKeyManagerUrlForPrivateKeys()!);
         Catch.setHandledTimeout(async () => {
           const { privateKeys } = await keyManager.getPrivateKeys(idToken);
           await processKeysFromEkm(acctEmail, privateKeys.map(entry => entry.decryptedPrivateKey), clientConfiguration, factory, idToken, ppEvent);
-        }, 0);
+        }, 0, completion);
+      } else {
+        completion();
```

Comment:
This would be a hanging promise?  I wonder why linter isn't giving warning about this.
Wouldn't it be more flexible to return the function in case it can't be consumed?
So `startPullingKeysFromEkm` accepts parameter `completion: () => Promise<void>`
and returns `Promise<Promise<void> | undefined>`?
So we can await the completion properly (if needed) or attach `catch` handler for it properly.